crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1316,6 +1317,25 @@ impl<T: Default> FromWorld for T {
     }
 }
 
+/// Wrapper type to enable getting multiple [`FromWorld`] types in one type.
+///
+/// This wrapper is necessary because a [`FromWorld`] implementation for a tuple of
+/// [`FromWorld`] implementing types would conflict with the [`FromWorld`] impl over all
+/// [`Default`] type.
+pub struct FromWorldWrap<T>(pub T);
+
+macro_rules! impl_from_world {
```

Comment:
Ping @Shatur, who wanted something similar in #4265.

Reply:
Thanks, looks interesting. But my issue was a conflict between my `Default` and `FromWorld` impl. So I think it solves a different problem.

Reply:
It's there to enable having multiple FromWorld elements in a more rigid API that might not allow listing more than one type. I'm pretty sure I had a concrete use case in mind when I first made this PR, but right now I can't remember any. Still, given that it's already implemented I see no reason not to keep it.

Reply:
> Still, given that it's already implemented I see no reason not to keep it

I don't like including code we can't imagine a use case for, especially in this case where its a little bit "odd". Additionally, because its a wrapper type, this could be done as a 3rd party library if someone actually needs it. My vote is to remove it unless we have a good reason to include it.